### PR TITLE
Fix acceptance tests on Arch and Ubuntu w/ Puppet 6

### PR DESCRIPTION
**Pull request by ekohl**
This doesn't take care of EL7, which I haven't looked into yet. But it's progress.

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

**Comment by root-expert** [comment]
Ubuntu specs will be fixed after https://github.com/puppetlabs/puppetlabs-apt/pull/1017 gets merged an released. Although we need to explicitly provide `release` parameter to `apt` or install `lsb-release` in our CI after we make some modulesnot using `os.distro.codename` fact.

File: manifests/params.pp [hunk 1]
```diff
@@ -112,7 +112,7 @@
       }
     }
     'Debian': {
-      if ($facts['os']['name'] == 'ubuntu' and $facts['os']['distro']['codename'] == 'xenial') {
+      if ($facts['os']['name'] == 'ubuntu' and $facts['os']['release']['major'] == '16.04') {
```
[the remark this answers is not in the document]
Comment (root-expert) [reply manifests/params.pp:115]: I have opened #1498 for this. I can close it if we include it in this PR.

**Review by bastelfreak — APPROVED** [review]

**Comment by ekohl** [comment]
I literally just wrote https://github.com/puppetlabs/puppetlabs-apt/pull/1018, but I see yours is more complete. I'll update this one.

**Comment by ekohl** [comment]
It looks like https://github.com/voxpupuli/puppet-nginx/pull/1473 now breaks EL7 :(

File: manifests/params.pp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ekohl) [reply manifests/params.pp]: Merged that one and rebased this PR.

File: spec/setup_acceptance_node.pp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+if $facts['os']['name'] == 'Ubuntu' {
+  # Facter < 4 needs lsb-release for os.distro.codename
+  if versioncmp($facts['facterversion'], '4.0.0') <= 0 {
+    package { 'lsb-release':
+      ensure => installed,
+    }
+  }
+}
```
Comment (root-expert) [inline spec/setup_acceptance_node.pp:8]: I'm not a fan of adding this to every single module that needs it.
At least if we do we should probably add it to a file that can be easily removed with modulesync?

Comment (ekohl) [reply spec/setup_acceptance_node.pp:8]: If it's common, perhaps it's something that should live in voxpupuli-acceptance? Right now I don't have time to work on that, but can review things if needed.

**Review by root-expert — APPROVED** [review]

**Pull request merged**